### ISSUE TITLE
[mu4e bug] "No buffer named *Summary*" when trying to save attachments

### ISSUE THREAD

**Issue by jgoerzen** [djcb/mu#2690]
**Describe the bug**

When using mu4e-view-mime-part-action, if I select a part and then hit s to save, I get:

No buffer named *Summary*

**How to Reproduce**

As above, use mu4e-view-mime-part-action, select a part, then hit s to save.  Backtrace is:

```
Debugger entered--Lisp error: (error "No buffer named *Summary*")
  gnus-article-part-wrapper(4 mm-save-part)
  gnus-article-save-part(4)
  #f(compiled-function (id) #<bytecode 0x18d1a58f937e8e50>)(4)
  mapc(#f(compiled-function (id) #<bytecode 0x18d1a58f937e8e50>) (4))
  seq-do(#f(compiled-function (id) #<bytecode 0x18d1a58f937e8e50>) (4))
  mu4e-view-mime-part-action()
  funcall-interactively(mu4e-view-mime-part-action)
  command-execute(mu4e-view-mime-part-action)
```

The same thing happens if I run `C-u 4 M-x gnus-article-save-part`.

Hitting o to open the part with the same number works fine.

**Environment**

Debian bookworm
Emacs 29.3
mu4e 1.12.3

**Checklist**
- [X] you are running either an 1.8.x/1.10.x release or `master` (otherwise please upgrade)
- [ ] you can reproduce the problem without 3rd party extensions (including Doom/Evil, various extensions etc.)  -- I do run under Doom and it is quite difficult to run mu4e without it; however, I don't see anything Doom-related in this backtrace.
- [X] you have read all of the above

Please make sure you all items in the checklist are set/met before filing the ticket. Thank you!

**Comment by djcb** [comment]
This has been fixed in `master`, soon-to-be 1.12.4.

**Comment by jgoerzen** [comment]
Thanks!

**Comment by djcb** [comment]
Fixed with 1.12.4.
